### ISSUE TITLE
[COMPAT] 32LK540BPSA webOS 4.3.0 (05.30.11) - Video not working

### ISSUE THREAD

**Issue by cybercuate** [mariotaku/moonlight-tv#35]
# Basic information

 - Device Model: 32LK540BPSA
 - System Version: 05.30.11
 - Is Audio Working: Yes
 - Is Video Working: No
 - Is Input Working: Remote control, wired mouse and keyboard, LG app through android phone

Tested on Wifi 2.4Ghz and Ethernet. Image appears showing 1 frame of the PC (static image). Input works as I click on many elements in the static image and on the PC things work (like clicking start menu or runing programs) but the picture on the LG TV keeps like a static image. Sound works well, and the app statistics shows up to 10ms delay over wifi 2.4Ghz and no frames dropped. Tried with different bitrates but no success. TV is able to stream very large movies (2K/60) through LAN or wifi without problem. WebOS version is 4.0 (2018)

**Comment by mariotaku** [comment]
After few updates, new version might solve that for you. You may choose one of the decoder, and there should be one for your TV.

**Comment by mariotaku** [comment]
Closed due to inactivity. Feel free to reopen if newest version doesn't work for you.

**Comment by mariotaku** [comment]
See others reporting similar issue

**Comment by mariotaku** [comment]
@Jessewb786 According to #51, they have similar device (OLED C8), NetCast Legacy isn't working for you either right?

**Comment by Jessewb786** [comment]
Netcast legacy isn't working for me either on the latest version. I could try give this version 6.6.0 a shot and see of that one works.

**Comment by Jessewb786** [comment]
@mariotaku I can confirm that video works on 0.6.6 on all resolutions and refresh rates.
I tested with NDL.
1080 60fps seemed ok.
1080 120fps (fps actually dropped in this mode) and 4k 60fps (choppy video) didn't work well even over a gigabit ethernet connection.

**Comment by mariotaku** [comment]
@Jessewb786 Thanks, this is very useful information. Could you please help me find which is the last working version?

**Comment by Jessewb786** [comment]
@mariotaku No problem, I'll run some tests later and let you know the results.

**Comment by mariotaku** [comment]
@Jessewb786 FYI, you can do test similar to binary testing. For example v0.7.11 -> v0.6.6 -> v0.7.0 -> v0.6.8

**Comment by Jessewb786** [comment]
@mariotaku Indeed, that would be the quickest way to zoom into the latest working version.

**Comment by Jessewb786** [comment]
@mariotaku It appears that my initial issue was wrong. In fact, with the latest version 0.7.11 and the NDL decoder, it seems to work well at 1080 120fps, but it stutters at 4k 60fps dropping down to 2fps even.

Note that it works fine at 4k 30 fps, so, I could be wrong, but I suspect that either 50000 kbps may be too little for this decoder at 4k 60fps, or perhaps the CPU just can't keep up (I hope that's not the case). I've seen that moonlight on Mac OS for example suggests 80000 kbps for 4k 60fps.

**Comment by mariotaku** [comment]
@Jessewb786 usually 35mbps is good enough and your TV will have hard time beyond that. Great it works anyway. I'll keep looking on this specific device. Thanks for your report!

**Comment by bnjroos** [comment]
Same issue here, sound but no picture, same firmware 05.30.11, different tv.

**Comment by raeldor** [comment]
How can I change the codec?  I'm also have the no video issue on 4.3.0.  Thanks.

**Comment by mariotaku** [comment]
@raeldor It's in Settings - Decoder Settings. Try NDL or NetCast Legacy. For Audio you can use automatic.

**Comment by raeldor** [comment]
Ah, this means changing/rebuilding the app?  It can't be done from the app itself?  Is compatibility with 4.30 going to be fixed in the next version?  Is anyone working on the root cause?  Thanks.

**Comment by sachamorgese** [comment]
I have the same problem with an OLED55B8SLC. Everything works except for the video, with the latest version. I haven't tried an older version yet.

My firmware is 05.30.20

**Comment by raeldor** [comment]
Tried again after latest software update (05.40.09) just a black screen after launching game now. :(

**Comment by mariotaku** [comment]
@raeldor Is it still webOS 4.3.0? 05.40.09 is firmware version and is not related to system version.

**Comment by Alejandro19851985** [comment]
@mariotaku  hi! first of all thank u for ur amazing job with the moonlight tv app for lg tv´s, im having the same problem like a lot of people on the coments i have an lg tv mod. 49lk5700psc web os ver. is 05.40.09 my gpu it´s gtx750 i´ve been getting black screen on lg tv but the game sound hears perfect the game pad seems to be working too, i think its a resolution thing or dont know if its a refresh rate thing or something like that, then i´ve treid to change the video decoder to ndl and im juste gettin an static single frame image monlight tv client app ver. is v0.9.6
[nvidia and internet hosting tool log.txt](https://github.com/mariotaku/moonlight-tv/files/7141160/nvidia.and.internet.hosting.tool.log.txt)

**Comment by mariotaku** [comment]
Some reports about firmware 05.40.09 (webOS 4.4.0): The video is not working

**Comment by lily-es** [comment]
OLED48A16LA, firmware 03.21.20. Video NDL, Audio NDL. Application version 1.4.0

Got a static image of my desktop when two monitors were plugged in to the PC. Audio fine, input fine.

Attempted disabling the second monitor (Show only on monitor 1 option in windows display settings), but noticed that when i started streaming a game, the pc would turn the second monitor back on. I saw it was actually displaying the second monitor's desktop, where there was nothing happening, so it looked like a static image. After unplugging the second monitor and trying again, instead of a static image I got a black screen. 

Restarting the computer with the second monitor unplugged got video working!
Might not be relevant to the issue, bot dropping this here anyway.

**Comment by HoodieHo** [comment]
Having the exact same issue - audio, controls ok, video - 1st frame of the pc. Tried different decoders/resolutions/fps in moonlight - no fix. Works fine when streaming to android.

model: 43LK5910PLC
webOS: 4.4.0-12 (goldilocks-gorce)

**Comment by 4tl4sh** [comment]
Having similar issue. 

Default settings works OK. Here's my testing.

**Model:** 43UK6400PLF
**webOS:** 4.4.0


<html>
<body>


**Decoder / Codec** | **Working resolutions** 
-- | --
**SMP h264** | None 
**SMP h265** | None
**NDL h264** | 720, 1080, 1440* 
**NDL h265** | Not compatible
**NetCast Legacy** | 720, 1080, 1440*


</body>
</html>

*1440 streams but with a green glitch screen. Only 720 and 1080 resolutions work properly at 30 or 60 fps allowing multiple bitrate settings.

**Comment by OutPut-translate** [comment]
Всем привет. После сбоя в телевизора появилась проблема с трансляцией, вылетает при соединении, перепробовал разные кодеки. Ранее до сброса телевизора до заводских настроек все работало. Единственное на что я могу грешить это обновленный драйвер видеокарты. Больше в системе ни чего не менялось вроде. Время будет, попробую откатиться по драйверам. Отпишусь как что.

Hi all.  After a failure on the TV, there was a problem with the broadcast, it crashes when connected, I tried different codecs.  Previously, before resetting the TV to factory settings, everything worked.  The only thing I can sin on is the updated video card driver.  Nothing else has changed in the system.  There will be time, I'll try to roll back the drivers.  I'll write it down like.

**Comment by mariotaku** [comment]
Update after 4 years(!):

I purchased main board of similar model, extracted from a broken TV. At first, Stream worked well, but not after I updated its firmware.

So I tried to forcefully downgrade, and the stream works again.

What's working:

* 05.10.05 (webOS 4.1.3-5915)

**Comment by mariotaku** [comment]
So... I took one step further, to pinpoint what breaks `libkadaptor`, and it turns out in the newer version, there is a bug that tells the system decoder to process H265 even when we're providing H264.

And I was able to create a fix that doesn't require any user action. In the next release, Moonlight would work even without replacing the `libkadaptor`.
